Limit-Theory-Redux/ltheory
# New github action scripts

This workflow does:
- Cancels last workflow of same type
- Builds binaries
- Delete old 'latest' release + tag
- Create new 'latest' release + tag
- Reads package version from root Cargo.toml
- Nightly builds (main on push): Sets app version in `script/Config/Version.lua` to [pkg-version]-[short_sha] e.g. `0.1.0-52b5775`
- Pull Requests: Sets app version in `script/Config/Version.lua` to [pkg-version]-[branchname]-[short_sha] e.g. `0.1.0-feature/print_hello_world-52b5775`
- workflow ignores certain files on push, see my comments in https://github.com/Limit-Theory-Redux/ltheory/pull/180/commits/fd9de6e4441c9b248cb4a6ca3cd661519906a1b8

A test release can be viewed on my [fork](https://github.com/IllustrisJack/ltheory-redux/releases/tag/latest)

Release / Tag:
![image](https://github.com/Limit-Theory-Redux/ltheory/assets/76592751/fb4faaf9-da20-46ca-addb-450597253d80)

In-Game (bottom-left):
![image](https://github.com/Limit-Theory-Redux/ltheory/assets/76592751/baf054d5-c4e6-48cc-a2c1-9fa9d5aff1fb)
![](https://user-images.githubusercontent.com/76592751/283641471-1d8c4777-ca66-4efe-9bb6-4b00995dd903.png)
